actionview/lib/action_view/helpers/tags/base.rb

```diff
@@ -148,7 +148,7 @@ def select_content_tag(option_tags, options, html_options)
               options[:include_blank] ||= true unless options[:prompt]
             end
 
-            value = options.fetch(:selected) { value(object) }
+            value = options.fetch(:selected) { value() }
```

Comment:
Let's remove `()`.

Reply:
We can not change to `value` here. If we change, 1 test case fails

```
Failure:
FormOptionsHelperTest#test_select_no_prompt_when_select_has_value [rails/actionview/test/template/form_options_helper_test.rb:717]:
Expected: <select id="post_category" name="post[category]"><option value="abe">abe</option>
<option value="&lt;mus&gt;" selected="selected">&lt;mus&gt;</option>
<option value="hest">hest</option></select>
Actual: <select name="post[category]" id="post_category"><option value="">Please select</option>
<option value="abe">abe</option>
<option selected="selected" value="&lt;mus&gt;">&lt;mus&gt;</option>
<option value="hest">hest</option></select>
```

`value` in `options.fetch(:selected) { value }` is interpreted as "local variable get", because `value` local variable is defined in this method.

We can change to `value`, if we change local variable name (e.g. to `selected`).  Which do you think is better?

```diff
diff --git a/actionview/lib/action_view/helpers/tags/base.rb b/actionview/lib/action_view/helpers/tags/base.rb
index 501cf29c95..0a00b58b61 100644
--- a/actionview/lib/action_view/helpers/tags/base.rb
+++ b/actionview/lib/action_view/helpers/tags/base.rb
@@ -148,8 +148,8 @@ def select_content_tag(option_tags, options, html_options)
               options[:include_blank] ||= true unless options[:prompt]
             end

-            value = options.fetch(:selected) { value() }
-            select = content_tag("select", add_options(option_tags, options, value), html_options)
+            selected = options.fetch(:selected) { value }
+            select = content_tag("select", add_options(option_tags, options, selected), html_options)

             if html_options["multiple"] && options.fetch(:include_hidden, true)
               tag("input", disabled: html_options["disabled"], name: html_options["name"], type: "hidden", value: "") + select
```

Reply:
I see. keep `()` then.